android/app/src/main/java/com/mattermost/rnbeta/NotificationReplyBroadcastReceiver.java

```diff
@@ -53,12 +41,7 @@ public void onReceive(Context context, Intent intent) {
             final int notificationId = intent.getIntExtra(CustomPushNotificationHelper.NOTIFICATION_ID, -1);
             final String serverUrl = bundle.getString("server_url");
             if (serverUrl != null) {
-                final ReactApplicationContext reactApplicationContext = new ReactApplicationContext(context);
-                final String token = Credentials.getCredentialsForServerSync(reactApplicationContext, serverUrl);
-
-                if (token != null) {
-                    replyToMessage(serverUrl, token, notificationId, message);
-                }
+                    replyToMessage(serverUrl, notificationId, message);
             } else {
                 onReplyFailed(notificationId);
```

Comment:
I'm not 100% here but is it creating an infinite loop?

Reply:
How?

Reply:
At line 46, `onReplyFailed` will invoke `recreateNotification`  which in turn will call `notificationManager.notify(notificationId, notification);` .  Now, if the `serverUrl ` was never in the `bundle`.  would this trigger a loop and always hit the `if block` at line 43?

Reply:
no cause `onReceive` is not being called after notify.. there are two different onReceive, this one which is when you reply directly in the notification and the `onReceive` when a remote notification arrives.

Feel free to give it a try

Reply:
Yes thanks :)